projects/saturn/src/main/java/io/fairspace/saturn/config/ViewsConfig.java

```diff
@@ -68,6 +80,12 @@ public static class View {
          * The name of the items that appear as rows.
          */
         public String itemName;
+        /**
+         * The max count value to be requested for a view total count.
+         * If total count is greater than this max value, the total count value will look like 'more than 1000' on FE.
+         * This is to prevent performance issues when the total count is too large.
+         */
+        public Long maxDisplayCount;
```

Comment:
Maybe with @Positive annotation? Then it needs to be > 0 or null.
```suggestion
        @Positive
        public Long maxDisplayCount;
```
We don't validate enough the values of this config, there is a ticket to improve it, but we can already start making it better. I guess the negative value of limit in Postgres doesn't make sense

Reply:
I would do it in the scope of this ticket https://thehyve.atlassian.net/browse/FAIRSPC-80. Now there are a few annotations like @NotBlank in the properties class, but they don't work, validation is not triggered anywhere. So, I would leave it for now, but I dropped it in the 80th ticket description